### ISSUE TITLE
Should install scripts get their own directory?

### ISSUE THREAD

**Issue by lillian-lemmer** [hypatia-engine/hypatia#61]
I'm not sure. They'd have to all begin with changing to the parent directory, then...

**Comment by ejmr** [comment]
If you decide to move them, you can use `git rev-parse --show-toplevel` to get the top-level directory of the project.  So for example, `install-ubuntu-python2.sh` has this line:

```sh
./install-base-python2.sh
```

If you moved `install-ubuntu-python2.sh` into a sub-directory (or deeper) then you could change that line to this:

```sh
BASE_INSTALL_SCRIPT="$(git rev-parse --show-toplevel)/install-base-python2.sh
exec "$BASE_INSTALL_SCRIPT"
```

That assumes `install-base-python2.sh` is still in the same top-level location it is now.  If you were to move it as well then you'd add the appropriate sub-directories after `git rev-parse --show-toplevel`.

**Comment by lillian-lemmer** [comment]
Great notes, @ejmr! I will adhere to your advice if I decide to indeed put install scripts in their own directory.

**Comment by alxcenz** [comment]
I'm questioning if the install scripts are strictly necessary. They could easily be replaced with a little documentation, which would be better than creating install scripts for every platform under the sun...

My 2c anyway.

**Comment by ejmr** [comment]
@alxcenz I agree that the installation scripts should be supplemented by documentation, explaining dependencies to install and things like that.  But it's always been my impression that the goal is to provide install scripts for commonly-used platforms, not *all* of them, since that would be overkill (which I felt like you were alluding to).

**Comment by lillian-lemmer** [comment]
Yeah, I mean honestly the biggest benefit is just... I have a lot of friends that are using Unix-y platforms and they're not Python developers and they wanna checkout my work, so I'm just like "Well just run the install script." :) I know I could be creating binaries, distributions, packages, etc., but that honestly takes a whole lot more of my time compared to maintaining the install scripts. In the future, however, @alxcenz has the right idea--Hypatia will be large enough to have actual dedicate platform-specific ports being developed separately and the install scripts will be completely redundant, save the ability to easily install nightlies, but who cares about that (I don't see any reason a non-developer should be playing with nightlies).

**Comment by alxcenz** [comment]
Closed with #69.
